### PR TITLE
[2.6] ENT-2725: Update system purpose status to NOT consider Layered SLA

### PR DESCRIPTION
**Pull request by sonalidhome**
- Updated the `rules.js` to ignore pool in priority calculation
 - Updated `SystemPurposeComplianceRules` to ignore

### CODE REVIEW

**Review by Ceiu — CHANGES_REQUESTED** [review]
Seems good, and looks like it follows the accepted design.

Just one oddity with the spec test and I'm okay with it.

File: server/spec/autobind_spec.rb [hunk 1]
```diff
@@ -682,4 +682,42 @@
...
+    expect(status['status']).to eq('not specified')
+
+    entitlements = @cp.list_entitlements(:uuid => consumer.uuid)
+
+    # Trying to bind the without SLA exempt.
+    # Then system purpose status should be match.
+
+    entitlements.each do |ent|
+      @cp.unbind_entitlement(ent.id, {:uuid => consumer.uuid})
+    end
```
[the remark this answers is not in the document]
Comment (sonalidhome) [reply server/spec/autobind_spec.rb:718]: Good catch! :) 
I updated the spec-test to test more stuff in.

[the remark this answers is not in the document]
Comment (sonalidhome) [reply server/spec/autobind_spec.rb:718]: Good catch! :)
I updated the spec-test to test more stuff in.

File: server/src/main/resources/rules/rules.js [hunk 2]
```diff
@@ -506,20 +506,25 @@ function get_pool_priority(pool, consumer) {
         var null_rule_score = 0;
         var mismatch_rule_score = 0;
 
-        log.debug("Evaluating attribute {} with weight {}", attr, attrs[attr]);
-        if (unsatisfiedSet.length === 0 && poolSet.length === 0) {
-            null_rule_score = 0.1;
-            log.debug("NULL rule score for attribute {} and pool {}, which is: {}", [attr, pool.id, null_rule_score]);
+        if(Utils.equalsIgnoreCase(attr, 'support_level') && Utils.equalsIgnoreCase(poolSet[0], 'Layered')) {
+            log.debug("Ignoring the pool prioritization when SLA is Layered", attr, poolSet[0]);
```
Comment (nikosmoum) [inline server/src/main/resources/rules/rules.js:510]: Similarly to the check for the syspurpose compliance, we shouldn't be checking by hardcoding `Layered` here, but instead checking if pool we are evaluating has the `support_level_exempt` attribute set to true. Something like `pool.retrievePoolAttributeValues("support_level_exempt")` should work here (although I have not tested it, so it needs verification).

**Review by nikosmoum — CHANGES_REQUESTED** [review]
One thing to change. Otherwise we're good. Also, after this is merged, let's also fix this in master

Edit: As discussed, let's remove the bz id from the commit/PR

File: server/src/main/resources/rules/rules.js (hunk 2 above)
Comment (nikosmoum) [reply server/src/main/resources/rules/rules.js:510]: UPDATE: based on the latest discussion, we should actually remove these rules.js changes all together

Comment (sonalidhome) [reply server/src/main/resources/rules/rules.js]: Removed the changes related to rules.js.

**Pull request merged**